code/go/0chain.net/miner/protocol_block_integration_tests.go

```diff
@@ -78,18 +81,64 @@ func hasDST(pb, b []*transaction.Transaction) (has bool) {
 	return false // has not
 }
 
+type ChallengeResponseTxData struct {
+	Name  string                      "json:'name'"
+	Input storagesc.ChallengeResponse "json:'input'"
+}
+
 /*UpdateFinalizedBlock - update the latest finalized block */
 func (mc *Chain) UpdateFinalizedBlock(ctx context.Context, b *block.Block) {
 	mc.updateFinalizedBlock(ctx, b)
 
-	if mc.isTestingOnUpdateFinalizedBlock(b.Round) {
+	addResultIfAdversarialValidatorTest(b)
+
+	if isTestingOnUpdateFinalizedBlock(b.Round) {
 		if err := chain.AddRoundInfoResult(mc.GetRound(b.Round), b.Hash); err != nil {
 			log.Panicf("Conductor: error while sending round info result: %v", err)
 		}
 	}
 }
 
-func (mc *Chain) isTestingOnUpdateFinalizedBlock(round int64) bool {
+func addResultIfAdversarialValidatorTest(b *block.Block) {
+	s := crpc.Client().State()
+
+	if s.AdversarialValidator == nil || !s.IsMonitor {
+		return
+	}
+
+	for _, tx := range b.Txns {
+		var txParsed map[string]interface{}
```

Comment:
do we need txParsed?

Reply:
fixed by [8843e24](https://github.com/0chain/0chain/pull/1603/commits/8843e249a2cc8643e5309411e150b118bfd53f4e)